Assets/JavaScript/script.js

```diff
@@ -0,0 +1,70 @@
+// Assignment Code
+var generateBtn = document.querySelector("#generate");
+
+function generatePassword() {
+  //Declare the constant variable
+  const lowercase = "abcdefghijklmnopqrstuvwxyxz";
+  const uppercase = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
+  const numbers = "0123456789";
+  const specials = "!@#$%^&*()?/;:~";
+
+  //Determine password length
+  const passwordLength = Number(prompt("Enter password length (8-128):"));
+
+  if (passwordLength < 8 || passwordLength > 128) {
+    alert("Please choose a different length?");
+    return passwordLength;
+  } else if (passwordLength >= 8 || passwordLength <= 128) {
+  }
+
+  //Functions to determine what user wants to include
+  const isLowerCase = confirm("Do you want to add a lowercase?");
+  const isUpperCase = confirm("Do you want to add a uppercase?");
+  const isNumbers = confirm("Do you want to add a number?");
+  const isSpecials = confirm("Do you want to add a special character?");
+
+  if (!isLowerCase && !isUpperCase && !isNumbers && !isSpecials) {
+    alert("please choose at least one option");
+    return;
+  }
+
+  // Return value
+  const choices = [];
+
+  if (isLowerCase) {
+    choices.push(lowercase);
+  }
+  if (isUpperCase) {
+    choices.push(uppercase);
+  }
+  if (isNumbers) {
+    choices.push(numbers);
+  }
+  if (isSpecials) {
+    choices.push(specials);
+  }
+
+  let password = "";
+  // generate random password
+  for (let i = 0; i < 10; i++) {
```

Comment:
Double check the condition of this for loop (the bit where you have stated `i < 10`. Why have you put `10`? And what effect does this have on your loop and resulting password?